tests/test_kubernetes.py

```diff
@@ -171,6 +173,9 @@ def setUp(self):
         self.mock_get_ep.content = '{"subsets":[{"ports":[{"name":"https","protocol":"TCP","port":443}],' +\
                                    '"addresses":[{"ip":"127.0.0.1"},{"ip":"127.0.0.2"}]}]}'
 
+        with patch('builtins.open', mock_open(read_data=json.dumps(config))):
+            k8s_config.load_kube_config()
+
```

Comment:
I am not sure this is necessary.
It is much simpler to just decorate `TestApiClient` with `@patch.object(K8sConfig, '_server', '', create=True)`.